src/Microsoft.Extensions.Http/DefaultHttpClientFactory.cs

```diff
@@ -344,5 +359,20 @@ public static void HandlerExpired(ILogger logger, string clientName, TimeSpan li
                 _handlerExpired(logger, lifetime.TotalMilliseconds, clientName, null);
             }
         }
+
+        // Wraps the inner handler in the same manner as HttpClient so that the caller can
+        // dispose of the handler freely without affecting it being shared with other clients
+        private sealed class LifetimeTrackingHttpMessageHandlerWrapper : DelegatingHandler
```

Comment:
I'd prefer if we could bake this behavior into `LifetimeTrackingHttpMessageHandler` rather than creating an extra object. It seems like we could do this there with no impact on the existing behavior. Thoughts?

Reply:
Sure, I can look into doing that.